File: Python-TriviaGame/TriviaGame.py
correct = 0
incorrect = 0
ans1 = 0
ans2 = 0
ans3 = 0
ans4 = 0

def linebreak():
    print ('\n')

while correct == 0 and incorrect == 0:
    print ("What is the answer to Question 1? Enter A, B or C")
    print ("A. CORRECT ANSWER")
    print ("B. Incorrect answer 1")
    print ("C. Incorrect answer 2")
    #This loop gives a user the option to restart the survey
    #Entering Y will set the stored values for the "correct" and "incorrect" variables, which meets the condition to start the quiz

    while ans1 != "A" or ans1 != "a" or ans1 != "B" or ans1 != "b" or ans1 != "C" or ans1 != "c":
    #The question will be repeated if the user enters an invalid value
    #If the value is not equal to A B or C then the program will require the user to re-enter a value until a valid answer is entered

        ans1 = input ("Answer: ")

        if ans1 == "A" or ans1 == "a":
                correct = correct + 1
                print (ans1, "is correct!")
                linebreak()
                break

        elif ans1 == "B" or ans1 == "b" or ans1 == "C" or ans1 == "c":
                incorrect = incorrect + 1
                print (ans1, "is not the correct answer.")
                linebreak()
                break

        else:
                print (ans1, "is not a valid entry")
                continue

    print ("What is the answer to Question 1? Enter A, B or C")
    print ("A. Incorrect answer 1")
    print ("B. CORRECT ANSWER")
    print ("C. Incorrect answer 2")

    while ans2 != "A" or ans2 != "a" or ans2 != "B" or ans2 != "b" or ans2 != "C" or ans2 != "c":

        ans2 = input ("Answer: ")

        if ans2 == "B" or ans2 == "b":
                correct = correct + 1
                print (ans2, "is correct!")
                linebreak()
                break

        elif ans2 == "A" or ans2 == "a" or ans2 == "C" or ans2 == "c":
                incorrect = incorrect + 1
                print (ans2, "is not the correct answer.")
                linebreak()
                break

        else:
                print (ans2, "is not a valid entry")
                continue
            
    print ("What is the answer to Question 1? Enter A, B or C")
    print ("A. CORRECT ANSWER")
    print ("B. Incorrect answer 1")
    print ("C. Incorrect answer 2")

    while ans3 != "A" or ans3 != "a" or ans3 != "B" or ans3 != "b" or ans3 != "C" or ans3 != "c":

        ans3 = input ("Answer: ")

        if ans3 == "A" or ans3 == "a":
                correct = correct + 1
                print (ans3, "is correct!")
                linebreak()
                break

        elif ans3 == "B" or ans3 == "b" or ans3 == "C" or ans3 == "c":
                incorrect = incorrect + 1
                print (ans3, "is not the correct answer.")
                linebreak()
                break

        else:
                print (ans3, "is not a valid entry")
                continue

    print ("What is the answer to Question 1? Enter A, B or C")
    print ("A. Incorrect answer 1")
    print ("B. Incorrect answer 2")
    print ("C. CORRECT ANSWER")

    while ans4 != "A" or ans4 != "a" or ans4 != "B" or ans4 != "b" or ans4 != "C" or ans4 != "c":

        ans4 = input ("Answer: ")

        if ans4 == "C" or ans4 == "c":
                correct = correct + 1
                print (ans4, "is correct!")
                linebreak()
                break

        elif ans4 == "A" or ans4 == "a" or ans4 == "B" or ans4 == "b":
                incorrect = incorrect + 1
                print (ans4, "is not the correct answer.")
                linebreak()
                break

        else:
                print (ans4, "is not a valid entry")
                continue

    print ("You got", correct, "of", correct + incorrect, "questions correct.")
    linebreak()
    #Provides a total score

    rerun = input ("Would you like to try again? Enter Y or N to continue: ")
    if rerun == "Y" or rerun == "y":
        incorrect = 0
        correct = 0
        print ("Welcome back!")
        linebreak()

    elif rerun == "N" or rerun == "n":
        linebreak()
        print ("Thanks for playing!")
        break

    else:
        print (rerun, "is not a valid entry")
        continue

File: CSharp-TriviaGame/TriviaGame.cs
using System;

namespace TriviaProject
{
    internal class Program
    {
        static void Main(string[] args)
        {

            var q1 = "x";
            var q1Corrected = "x";
            var q2 = "x";
            var q2Corrected = "x";
            var q3 = "x";
            var q3Corrected = "x";
            var q4 = "x";
            var q4Corrected = "x";
            double totalCorrect = 0.0;
            double totalIncorrect = 0.0;
            double finalScore = 0.0;
            var tryAgain = "Y";
            var tryAgainCorrected = "Y";

            while (tryAgainCorrected == "Y")
            {
                Console.WriteLine("Question 1: What is the answer?");
                Console.WriteLine("A. Correct Answer");
                Console.WriteLine("B. Incorrect Answer");
                Console.WriteLine("C. Incorrect Answer");

                while (q1Corrected != "A" && q1Corrected != "B" && q1Corrected != "C")
                {
                    q1 = Console.ReadLine();
                    q1Corrected = q1.ToUpper();

                    switch (q1Corrected)
                    {
                        case "A":
                            Console.WriteLine("{0} is the correct Answer!", q1Corrected);
                            ++totalCorrect;
                            break;

                        case "B":
                            Console.WriteLine("{0} is not correct.", q1Corrected);
                            ++totalIncorrect;
                            break;

                        case "C":
                            Console.WriteLine("{0} is not correct.", q1Corrected);
                            ++totalIncorrect;
                            break;

                        default:
                            Console.WriteLine("{0} is not a valid entry. A, B or C.", q1Corrected);
                            continue;
                    }
                }

                Console.WriteLine("Question 2: What is the answer?");
                Console.WriteLine("A. Incorrect Answer");
                Console.WriteLine("B. Correct Answer");
                Console.WriteLine("C. Incorrect Answer");

                while (q2Corrected != "A" && q2Corrected != "B" && q2Corrected != "C")
                {
                    q2 = Console.ReadLine();
                    q2Corrected = q2.ToUpper();

                    switch (q2Corrected)
                    {
                        case "A":
                            Console.WriteLine("{0} is not correct.", q2Corrected);
                            ++totalIncorrect;
                            break;

                        case "B":
                            Console.WriteLine("{0} is the correct Answer!", q2Corrected);
                            ++totalCorrect;
                            break;

                        case "C":
                            Console.WriteLine("{0} is not correct.", q2Corrected);
                            ++totalIncorrect;
                            break;

                        default:
                            Console.WriteLine("{0} is not a valid entry. A, B or C.", q2Corrected);
                            continue;
                    }
                }

                Console.WriteLine("Question 3: What is the answer?");
                Console.WriteLine("A. Incorrect Answer");
                Console.WriteLine("B. Incorrect Answer");
                Console.WriteLine("C. Correct Answer");

                while (q3Corrected != "A" && q3Corrected != "B" && q3Corrected != "C")
                {
                    q3 = Console.ReadLine();
                    q3Corrected = q3.ToUpper();

                    switch (q3Corrected)
                    {
                        case "A":
                            Console.WriteLine("{0} is not correct.", q3Corrected);
                            ++totalIncorrect;
                            break;

                        case "B":
                            Console.WriteLine("{0} is not correct.", q3Corrected);
                            ++totalIncorrect;
                            break;

                        case "C":
                            Console.WriteLine("{0} is the correct Answer!", q3Corrected);
                            ++totalCorrect;
                            break;

                        default:
                            Console.WriteLine("{0} is not a valid entry. A, B or C.", q3Corrected);
                            continue;
                    }
                }

                Console.WriteLine("Question 4: What is the answer?");
                Console.WriteLine("A. Incorrect Answer");
                Console.WriteLine("B. Incorrect Answer");
                Console.WriteLine("C. Correct Answer");

                while (q4Corrected != "A" && q4Corrected != "B" && q4Corrected != "C")
                {
                    q4 = Console.ReadLine();
                    q4Corrected = q4.ToUpper();

                    switch (q4Corrected)
                    {
                        case "A":
                            Console.WriteLine("{0} is not correct.", q4Corrected);
                            ++totalIncorrect;
                            break;

                        case "B":
                            Console.WriteLine("{0} is not correct.", q4Corrected);
                            ++totalIncorrect;
                            break;

                        case "C":
                            Console.WriteLine("{0} is the correct Answer!", q4Corrected);
                            ++totalCorrect;
                            break;

                        default:
                            Console.WriteLine("{0} is not a valid entry. A, B or C.", q4Corrected);
                            continue;
                    }
                }

                finalScore = (totalCorrect / (totalCorrect + totalIncorrect) * 100); 
                Console.WriteLine("You got {0} correct and {1} incorrect. Your final score is: {2}%.", totalCorrect, totalIncorrect, finalScore);
                Console.WriteLine("Would you like to try again? Enter Y to continue. Otherwise, enter any other value to Exit");

                tryAgain = Console.ReadLine();
                tryAgainCorrected = tryAgain.ToUpper();

                switch (tryAgainCorrected)
                {
                    case "Y":
                        Console.WriteLine("Welcome Back!");
                        q1 = "x";
                        q1Corrected = "x";
                        q2 = "x";
                        q2Corrected = "x";
                        q3 = "x";
                        q3Corrected = "x";
                        q4 = "x";
                        q4Corrected = "x";
                        totalCorrect = 0;
                        totalIncorrect = 0;
                        break;

                    default:
                        Console.WriteLine("Good bye!");
                        break;
                }
            }
        }
    }
} 